### PR TITLE
[ci] Remove unnecessary Linux libraries

### PR DESCRIPTION
**Pull request by stuartmorgan**
The Linux desktop template included some libraries as explicit dependencies unnecessarily, which were later removed from the template once we realized `snap` support didn't actually require them. This removes them from the Dockerfile, and from the CMakeLists.txt that still reference them.

Fixes https://github.com/flutter/flutter/issues/78562

### CODE REVIEW

**Comment by stuartmorgan** [comment]
No changelog needed: cruft removal in example build files isn't relevant to clients.

File: .ci/Dockerfile [hunk 1]
```diff
@@ -23,7 +23,7 @@ RUN apt-get install -y clang-format
 # - build tools.
 RUN apt-get install -y clang cmake ninja-build file pkg-config
 # - libraries.
-RUN apt-get install -y libgtk-3-dev libblkid-dev liblzma-dev libgcrypt20-dev
```
Comment (ditman) [inline .ci/Dockerfile:26]: sch-rdbl-lib-nms!

**Review by ditman — APPROVED** [review]
If CI is happy, hit it!

**Review by cbracken — APPROVED** [review]

**Comment by stuartmorgan** [comment]
The CI failure is just the OOB publishing failure that has the tree closed.

**Review by tarrinneal — APPROVED** [review]

**Pull request merged**